Assets/Tests/InputSystem/CoreTests_Actions.cs

```diff
@@ -10230,66 +10232,198 @@ public void Actions_WithMultipleCompositeBindings_WithoutEvaluateMagnitude_Works
         Assert.That(values[0].Position, Is.EqualTo(new Vector2(1, 1)));
     }
 
-    // FIX: This test is currently checking if shortcut support is enabled by testing that the unwanted behaviour exists.
-    // This test should be repurposed once that behaviour is fixed.
     [Test]
     [Category("Actions")]
-    [TestCase(true)]
-    [TestCase(false)]
-    public void Actions_ImprovedShortcutSupport_ConsumesWASD(bool shortcutsEnabled)
+    public void Actions_WithCompositeBindings_WithHandleInputEventsSetToTrue_HandleInputEvents()
     {
-        InputSystem.settings.shortcutKeysConsumeInput = shortcutsEnabled;
+        var keyboard = InputSystem.AddDevice<Keyboard>();
+
+        var map = new InputActionMap("map1");
+        var action1 = map.AddAction(name: "action1");
+        // 1D axis composite doesn't by default handle input events, but by setting handleInputEvent to true, it should
+        action1.AddCompositeBinding("1DAxis(handleInputEvents=true)")
+            .With("negative", "<Keyboard>/a")
+            .With("positive", "<Keyboard>/d");
+        var action2 = map.AddAction("action2", binding: "<Keyboard>/a");
 
+        var wasHandled = false;
+        action2.performed += ctx =>
+        {
+            wasHandled = ctx.eventHandled;
+        };
+
+        map.Enable();
+
+        Press(keyboard.aKey);
+
+        Assert.That(wasHandled, Is.True);
+    }
+
+    [Test]
+    [Category("Actions")]
+    public void Actions_ActionCallbacksCanSetEventsAsHandled()
+    {
         var keyboard = InputSystem.AddDevice<Keyboard>();
 
-        var map1 = new InputActionMap("map1");
```

Comment:
It isn't really relevant anymore because the composites in that test don't by default handle input events, so they never get consumed. Those paths are tested by other tests though.